### ISSUE TITLE
Enabling kubeflow on microk8s

### ISSUE THREAD

**Issue by TosinAina** [canonical/bundle-kubeflow#289]
I installed microk8s using `sudo snap install microk8s --classic`,
I enabled dns, dashboard and storage with `microk8s enable dns dashboard storage`,
I tried enabling kubeflow with `microk8s enable kubeflow`, inputted the password and it prompted me to add `--ignore-min-mem`,
I tried enabling kubeflow with `microk8s enable kubeflow --ignore-min-mem`, inputted the password and it gave me this error message 

Kubeflow could not be enabled:
error: deployment "calico-kube-controllers" exceeded its progress deadline
Command '('microk8s-kubectl.wrapper', '-nkube-system', 'rollout', 'status', 'deployment.apps/calico-kube-controllers')' returned non-zero exit status 1

I expected kubeflow to be enabled and also get a link to access the dashboard

**Environment:**

- Kubernetes platform: microk8s
- Kubernetes version: (use `kubectl version`): Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.2", GitCommit:"faecb196815e248d3ecfb03c680a4507229c2a56", GitTreeState:"clean", BuildDate:"2021-01-15T02:50:16Z", GoVersion:"go1.15.6", Compiler:"gc", Platform:"linux/amd64"}
- OS (e.g. from `/etc/os-release`): ubuntu 20.04 focal

**Comment by knkski** [comment]
@ktsakalozos: Any idea what might be the issue here?

**Comment by ktsakalozos** [comment]
Hi @TosinAina would you me able to run `microk8s inspect` and attach the gathered logs?

For some reason the calico CNI was not setup on time.

**Comment by TosinAina** [comment]
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-control-plane-kicker is running
  Service snap.microk8s.daemon-proxy is running
  Service snap.microk8s.daemon-kubelet is running
  Service snap.microk8s.daemon-scheduler is running
  Service snap.microk8s.daemon-controller-manager is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster
Inspecting juju
  Inspect Juju
Inspecting kubeflow
  Inspect Kubeflow

WARNING:  Docker is installed. 
File "/etc/docker/daemon.json" does not exist. 
You should create it and add the following lines: 
{
    "insecure-registries" : ["localhost:32000"] 
}
and then restart docker with: sudo systemctl restart docker
Building the report tarball
  Report tarball is at /var/snap/microk8s/1975/inspection-report-20210121_195748.tar.gz

**Comment by knkski** [comment]
@TosinAina: Could you upload the generated tarball to this thread (`/var/snap/microk8s/1975/inspection-report-20210121_195748.tar.gz`)?

**Comment by dinever** [comment]
Same error here

**Comment by knkski** [comment]
@dinever: Could you run `microk8s inspect` and attach the generated tarball?

**Comment by elgiewindhu** [comment]
Hi @knkski, I 'm facing the same issue here, my microk8s version is 1.20/stable. I uploaded the tarball here.

[inspection-report-20210610_095430.tar.gz](https://github.com/canonical/bundle-kubeflow/files/6629546/inspection-report-20210610_095430.tar.gz)

And I' facing a syntax error, everytime I try to disable kubeflow (I tried as suggestion to disable and enable kubeflow):

```
ubuntu@xxxx:~$ microk8s disable kubeflow
  File "/snap/microk8s/2213/scripts/wrappers/common/../../../actions/disable.kubeflow.sh", line 52
    click.echo(f"Destroying Kubeflow {resource}...")
                                                  ^
SyntaxError: invalid syntax
```

**Comment by elgiewindhu** [comment]
I forgot to apply the NO_PROXY on var/snap/microk8s/current/args/containerd-env, while I set the HTTP-PROXY behind my coorporate firewall. 🤦 

![image](https://user-images.githubusercontent.com/64198668/121998898-dfef4100-cdac-11eb-801a-7bf9c697f2ac.png)

So I tried to get the information about this 'calico-kube-control'-container with `microk8s kubectl describe all`  and apparently the container couldn't be created, because there was no direct connection to 10.152.183.1. After setting up the NO_PROXY on var/snap/microk8s/current/args/containerd-env, it worked very well then and this 'calico-kube-control'-container can be created 😄 

But after I got this solution, I got another https://github.com/ubuntu/microk8s/issues/1439 problem. 🤦
